src/clj_kondo/impl/linters/keys.clj

```diff
@@ -2,18 +2,64 @@
   {:no-doc true}
   (:require
    [clj-kondo.impl.findings :as findings]
-   [clj-kondo.impl.utils :refer [node->line tag]]))
+   [clj-kondo.impl.utils :refer [node->line tag]]
+   [clojure.string :as string]))
+
+(defn- ensure-no-nils
+  "Returns nil if the collection contains a nil, else returns coll.
+  Doesn't search for nested nils."
+  [coll]
+  (cond
+    (set? coll) (if (contains? coll nil)
+                  nil
+                  coll)
+    (sequential? coll) (if (contains? (set coll) nil)
+                         nil
+                         coll)
+    (map? coll) (if (contains? (set (flatten (seq coll))) nil)
+                  nil
+                  coll)
+    :else coll))
 
 (defn key-value
-  "We only support tokens as key values for now."
+  "We only support the following cases for now."
   [node]
   (case (tag node)
     :token (or (when-let [v (:k node)]
                  (if (:namespaced? node)
                    (str v) v))
                (str node))
+    :vector (ensure-no-nils (map key-value (:children node)))
```

Comment:
Can you try to write it in a short-cutting fashion, so we don't iterate more than needed?

E.g.: (pseudocode): `(reduce (fn [acc child] (if-let [v (key-value child)] ... (reduced nil)) (:children node))`

Reply:
Addressed in the latest commit.